Upstream has accepted an equivalent patch to the renamed test case:
https://github.com/mesonbuild/meson/commit/67d1948005579a59125ecb0340cecd62642510a0

From f8a86a92a779db53c37ca394ff04f1c9c87c98b7 Mon Sep 17 00:00:00 2001
From: Alan Coopersmith <alan.coopersmith@oracle.com>
Date: Sat, 29 Jun 2019 17:04:43 -0700
Subject: [PATCH] test cases/common/137 get define: add sunos

Signed-off-by: Alan Coopersmith <alan.coopersmith@oracle.com>
---
 test cases/common/137 get define/meson.build | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/test cases/common/137 get define/meson.build b/test cases/common/137 get define/meson.build
index c3364b93..34d96518 100644
--- a/test cases/common/137 get define/meson.build	
+++ b/test cases/common/137 get define/meson.build	
@@ -38,6 +38,9 @@ foreach lang : ['c', 'cpp']
   elif host_system == 'gnu'
     d = cc.get_define('__GNU__')
     assert(d == '1', '__GNU__ value is @0@ instead of 1'.format(d))
+  elif host_system == 'sunos'
+    d = cc.get_define('__sun')
+    assert(d == '1', '__sun value is @0@ instead of 1'.format(d))
   else
     error('Please report a bug and help us improve support for this platform')
   endif
-- 
2.19.2

